homeassistant/components/switch/fritzbox_callforwarding.py

```diff
@@ -0,0 +1,180 @@
+"""
+Support for FRITZ!Box call forwarding on/off.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.fritzbox_callforwarding/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['fritzconnection==0.6.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = '169.254.1.1'  # IP valid for all Fritz!Box routers
+DEFAULT_PORT = 49000
+DEFAULT_USERNAME = 'admin'
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=60)
+ICON = 'mdi:phone-forward'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_PASSWORD, default=''): cv.string,
+     vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+     vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string})
+
+
+class CallForwarding:
+    """Representation of call forwarding entity."""
+
+    def __init__(self, fritz_box, uid, from_number,
+                 to_number, connection_type, enabled):
+        """Initialize the call forwarding entity."""
+        self.fritz_box = fritz_box
+        self.name = "callforwarding_" + uid
+        self.uid = uid
+        self.from_number = from_number
+        self.to_number = to_number
+        self.connection_type = connection_type
+        self.enabled = enabled
+
+    def enable(self):
+        """Enable the call forwarding for a uid."""
+        kwargs = {'NewDeflectionId': self.uid, 'NewEnable': 1}
+        self.fritz_box.call_action(
+            'X_AVM-DE_OnTel:1', 'SetDeflectionEnable', **kwargs)
```

Comment:
This is very protocol specific. Can this be added to `FritzConnection` dependency?